components/director/internal/domain/package/repository.go

```diff
@@ -118,6 +123,34 @@ func (r *pgRepository) GetForApplication(ctx context.Context, tenant string, id
 	return pkgModel, nil
 }
 
+func (r *pgRepository) GetByInstanceAuthID(ctx context.Context, tenant string, instanceAuthID string) (*model.Package, error) {
+	var pkgEnt Entity
+
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	stmt := fmt.Sprintf(`SELECT %s FROM %s WHERE tenant_id=$1 AND id=(SELECT %s FROM %s WHERE tenant_id=$1 AND id=$2)`,
```

Comment:
What about such SQL:
`SELECT packages-cols FROM PACKAGES AS P JOIN PACKAGE_INSTANCE_AUTHS AS A on A.PACKAGE_ID=p.ID WHERE A.ID=instanceAuthID`
But generally it's ok.